source/common/upstream/original_dst_cluster.h

```diff
@@ -117,7 +136,6 @@ class OriginalDstCluster : public ClusterImplBase {
   Event::Dispatcher& dispatcher_;
   const std::chrono::milliseconds cleanup_interval_ms_;
   Event::TimerPtr cleanup_timer_;
-  const bool use_http_header_;
```

Comment:
nit: all of this new indirection seems unnecessary. Maybe just make this `absl::optional<Http::LowerCaseString> header_name_` to indicate whether to use header lookup and what header?

Reply:
Yeah, this start with a oneof but end up with this simple api.